nifi-nar-bundles/nifi-cassandra-bundle/nifi-cassandra-processors/src/main/java/org/apache/nifi/processors/cassandra/QueryCassandra.java

```diff
@@ -269,11 +313,17 @@ public void process(final OutputStream out) throws IOException {
             // cap the error limit at 10, format the messages, and don't include the stack trace (it is displayed by the
             // logger message above).
             getLogger().error(nhae.getCustomMessage(10, true, false));
+            if (fileToProcess == null) {
+                fileToProcess = session.create();
+            }
             fileToProcess = session.penalize(fileToProcess);
             session.transfer(fileToProcess, REL_RETRY);
-
         } catch (final QueryExecutionException qee) {
+            //session.rollback();
```

Comment:
There are a few spots of commented out code, please remove if not necessary, thanks!